### DISCUSSION TITLE
Why defaultValue or value api in Select component always must be string type ?

### DISCUSSION THREAD

**Discussion by iamyahia** [radix-ui/primitives#2612] [Help]
Hey developers,

I was checking out the radix-ui [documentation API](https://www.radix-ui.com/primitives/docs/components/select#root), and it mentions that both `defaultValue` and `value` types always have to be strings. But I'm curious why it's necessary for them to always be strings ?

In my project, I'm using the `Select component` from radix, and my `options value` looks like this:

```js
options = {
  value: string | number,
  label: string
}
```

In our project, the  `value key `sometimes has a `string value`, and in other cases, it might have a `number value`. Occasionally, we pass this value key to `defaultValue`.

However, the problem arises when we pass a number to it. TypeScript shows a warning message:

```js
Type 'number' is not assignable to type 'string'.ts(2322)
index.d.mts(13, 5): The expected type comes from property 'defaultValue' which is declared here on type 'IntrinsicAttributes & SelectProps'
```
So, my question remains: Why is it necessary for `defaultValue` and `value` to **always** be of type string? for what reason ?  for what logics ? i want to understand the purpos.

**Comment by benoitgrelard** [comment]
Because this is the underlying type of a native select.
